### PR TITLE
[PROJ4] Add search path for sqlite.exe

### PR DESCRIPTION
**Pull request by larshg**
**Describe the pull request**
Added additional search path.

- What does your PR fix? Fixes issue #10913

- Which triplets are supported/not supported? Have you updated the CI baseline?
Should support all triplets, also custom ones. Baseline not updated.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Does not seem like the right fix. You are forgetting ARM builds. The ARM builds are building on an x86/x64 host and cannot run the sqlite.exe from the arm triplet. So this fix might work for your custom triplet but fails with the windows cross triplets.

**Comment by larshg** [comment]
So, can this be solved by looking at the VCPKG_TARGET_ARCHITECTURE and make a special lookup into the x86-windows dir, if it contains anything with arm?

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also bump the version in CONTROL file?

**Comment by larshg** [comment]
Instead of modifying the existing search paths I just added the search path for the actual triplet folder. This should ensure that it is possible to build for arm/arm64 as before.

Doesn't it matter if the sqlite3 is x86 or x64 when generating the database to arm/arm64, since the FATAL message is specifically requesting the x86-windows?

**Comment by JackBoosY** [comment]
See #8763.

**Comment by larshg** [comment]
But the suggestion present in the [comment](https://github.com/microsoft/vcpkg/issues/8757#issuecomment-546795601). Is not the same thats in the master?

This check would have to be present to work for custom triplets.
```
if (EXISTS ${CURRENT_INSTALLED_DIR}/tools/sqlite3.exe)
            set(SQLITE3_BIN_PATH ${CURRENT_INSTALLED_DIR}/tools)

```

Currently all the testing, use explicit paths with specific triplet.

I guess I can install sqlite[tool] with a x86 triplet first, and the run my custom install triplets. Then it will pickup the x86 version of sqlite, but not my custom one,

@jasjuang does master really work for you know, with custom triplet?

File: ports/proj4/portfile.cmake [hunk 1]
```diff
@@ -33,6 +33,8 @@ if ("database" IN_LIST FEATURES)
             set(SQLITE3_BIN_PATH ${CURRENT_INSTALLED_DIR}/../x64-windows/tools)
         elseif (EXISTS ${CURRENT_INSTALLED_DIR}/../x64-windows-static/tools/sqlite3.exe AND (NOT CMAKE_HOST_SYSTEM_PROCESSOR OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64"))
             set(SQLITE3_BIN_PATH ${CURRENT_INSTALLED_DIR}/../x64-windows-static/tools)
+        elseif (EXISTS ${CURRENT_INSTALLED_DIR}/tools/sqlite3.exe)
```
Comment (JackBoosY) [inline ports/proj4/portfile.cmake:36]:
```suggestion
        elseif (NOT TRIPLET_SYSTEM_ARCH STREQUAL "arm" AND EXISTS ${CURRENT_INSTALLED_DIR}/tools/sqlite3.exe)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**